mipengine/controller/celery_app.py

```diff
@@ -1,20 +1,178 @@
+        super().__init__(message)
+
+
+class CeleryWrapper:
+    def __init__(self, socket_addr: str):
+        self._socket_addr = socket_addr
+        self._celery_app = self._create_new_celery_app()
+
+    def _close(self):
+        self._celery_app.close()
+
+    # queue_task() is non-blocking, because apply_async() is non-blocking
+    def queue_task(self, task_signature: str, *args, **kwargs) -> AsyncResult:
+        try:
+            task_signature = self._celery_app.signature(task_signature)
+            async_result = task_signature.apply_async(args, kwargs)
+            return async_result
+        except (
+            kombu.exceptions.OperationalError,
+            amqp.exceptions.AccessRefused,
+            amqp.exceptions.NotAllowed,
+        ) as exc:
+            # TODO how are things supposed to be logged in here?
+            # get_backgroung_service_logger() makes no sense
+
+            ######################
+            print(f"(celery_app::CeleryWrapper::queue_task) {exc=}\n")
+
+            import traceback
+
+            tr = traceback.format_exc()
+            print(tr)
+            ######################
+
+            self._close()
+            self._celery_app = self._create_new_celery_app()
+
+            connection_error = CeleryConnectionError(
+                connection_address=self._socket_addr,
+                error_details=f"while queuing {task_signature=} with {args=} and {kwargs=}",
+            )
+
+            raise connection_error
+
+    # get_result() is blocking, because celery.result.AsyncResult.get() is blocking
+    def get_result(
+        self, async_result: AsyncResult, timeout: int
+    ) -> Union[str, dict, list]:
+        try:
+            result = async_result.get(timeout)
+            return result
+        except (
+            celery.exceptions.TimeoutError,
+            billiard.exceptions.SoftTimeLimitExceeded,
+            billiard.exceptions.TimeLimitExceeded,
+        ) as timeout_error:
+            # TODO how are things supposed to be logged in here?
+            # get_backgroung_service_logger() makes no sense
+
+            ######################
+            print(
+                f"(celery_app::CeleryWrapper::get_result) {timeout_error=} {async_result.id=}"
+            )
+            import traceback
+
+            tr = traceback.format_exc()
+            print(tr)
+            ######################
+
+            try:
+                self._celery_app.control.inspect().ping()
+            except kombu.exceptions.OperationalError as oper_err_inner:
+
+                ######################
+                print(
+                    f"(celery_app::CeleryWrapper::get_result) {oper_err_inner=} {async_result.id=}"
+                )
+
+                import traceback
+
+                tr = traceback.format_exc()
+                print(tr)
+                ######################
```

Comment:
`queue_task` and `get_result` now instantiate a `controller_logger` inside their body using the `request_id'. `queue_task` already has the `request_id` in the `kwargs` and added the `request_id` as a input parameter in `get_result` but as I ve mentioned before the implementation of the `controller_logger` is problematic

Reply:
Looks good. Just a note here. You could pass the logger in the `CeleryWrapper` `__init__` method instead of passing the `request_id` each time in the methods.

That way, the `celery_app` wouldn't have a dependency on the `controller_logger` module, hence it wouldn't need a mock of the module if you needed to show the logs. You would just instantiate a logger and pass it in the object.

Related to: https://team-1617704806227.atlassian.net/browse/MIP-590

Reply:
changed to passing Logger object in `queue_task` and `get_result`